scripts/variantstore/variant_annotations_table/GvsValidateVAT.wdl

```diff
@@ -914,5 +932,207 @@ task ClinvarSignificance {
+        COUNT (DISTINCT vid) AS count FROM
+        (
+            SELECT vid
+            FROM `aou-genomics-curation-prod.beta2_99k.vat_v3`
+            WHERE exon_number IS NULL
+            AND aa_change IS NOT NULL
+            AND NOT (("intron_variant" IN UNNEST(consequence) AND (variant_type IN ("insertion", "deletion"))))
+            AND "splice_region_variant" NOT IN UNNEST(consequence)
+            AND "splice_acceptor_variant" NOT IN UNNEST(consequence)
+            AND "splice_donor_variant" NOT IN UNNEST(consequence)
+            AND "NMD_transcript_variant" NOT IN UNNEST(consequence)
+            AND "coding_sequence_variant" NOT IN UNNEST(consequence)
+        UNION DISTINCT
+            SELECT vid
+            FROM `aou-genomics-curation-prod.beta2_99k.vat_v3`
+            WHERE exon_number IS NOT NULL
+            AND aa_change IS NULL
+            AND "non_coding_transcript_exon_variant" NOT IN UNNEST(consequence)
+            AND "3_prime_UTR_variant" NOT IN UNNEST(consequence)
+            AND "5_prime_UTR_variant" NOT IN UNNEST(consequence)
+            AND "splice_donor_variant" NOT IN UNNEST(consequence)
+            AND "splice_acceptor_variant" NOT IN UNNEST(consequence)
+            AND "splice_region_variant" NOT IN UNNEST(consequence)
+            AND ("stop_gained" NOT IN UNNEST(consequence) AND "frameshift_variant" NOT IN UNNEST(consequence))
+            AND ("stop_lost" NOT IN UNNEST(consequence) AND "frameshift_variant" NOT IN UNNEST(consequence))
+            AND ("stop_lost" NOT IN UNNEST(consequence) AND "inframe_deletion" NOT IN UNNEST(consequence))
+            AND "NMD_transcript_variant" NOT IN UNNEST(consequence)
+            AND "stop_retained_variant" NOT IN UNNEST(consequence)
+            AND "incomplete_terminal_codon_variant" NOT IN UNNEST(consequence)
+            AND "transcript_variant" NOT IN UNNEST(consequence)
+            AND "coding_sequence_variant" NOT IN UNNEST(consequence)
+            AND "inframe_insertion" NOT IN UNNEST(consequence)
+            AND "inframe_deletion" NOT IN UNNEST(consequence)
+            AND "frameshift_variant" NOT IN UNNEST(consequence)
+            AND "mature_miRNA_variant" NOT IN UNNEST(consequence)
+        )' > bq_aachange_exonnumber.csv
+
+        NUMVARS=$(python3 -c "csvObj=open('bq_aachange_exonnumber.csv','r');csvContents=csvObj.read();print(csvContents.split('\n')[1]);")
+
+        # if the result of the bq call and the csv parsing is a series of digits, then check that it isn't 0
+        if [[ $NUMVARS =~ ^[0-9]+$ ]]; then
+            if [[ $NUMVARS = "0" ]]; then
+                echo "PASS: The VAT table ~{fq_vat_table} has consistency across all aa_change and exon_number values in it." > validation_results.txt
+            else
+                echo "FAIL: The VAT table ~{fq_vat_table} has $NUMVARS variants for which aa_change and exon_number are inconsistent." > validation_results.txt
```

Comment:
No, the way the code is written all of the validation tasks just put their output into a results file. The workflow itself does not fail if it fails validation. It might make sense to do so, but I'm following the pattern here that @RoriCremer is using.

Reply:
Definitely curious to hear what @RoriCremer thinks, personally I like validations that make some obvious noise when things don't check out. 🙂

Reply:
Agreed---I like obvious failures, but I'd prefer that _all_ the tests run